src/components/o-testimonials/js/testimonials.js

```diff
@@ -0,0 +1,163 @@
+import Swipe from '../../../js/swipe';
+import on from '../../../js/on';
+import UiEvents, { EVENTS } from '../../../js/ui-events';
+
+class Testimonials extends UiEvents {
+  static DEFAULTS = {
+    controlLeft: '.js-o-testimonials__control-left',
+    controlRight: '.js-o-testimonials__control-right',
+    slides: '.js-o-testimonial__item',
+    slider: '.js-o-testimonials',
+  };
+
+  constructor(wcNode, options = {
+  }) {
+    // eslint-disable-next-line no-param-reassign
+    options = {
+      ...Testimonials.DEFAULTS,
+      ...options,
+    };
+
+    super(wcNode, options);
+
+    this.slideIndex = 0;
+    this.autoRotateEnabled = true;
+    this.autoRotateTimeInMiliseconds = 5000;
+    this.options = options;
+    this.wcNode = wcNode;
+
+    this.init();
+  }
+
+  init() {
+    this.slides = this.wcNode.querySelectorAll(this.options.slides);
+    this.controlLeft = this.wcNode.querySelector(this.options.controlLeft);
+    this.controlRight = this.wcNode.querySelector(this.options.controlRight);
+    this.slider = this.wcNode.querySelector(this.options.slider);
+    this.autoRotateTimeInMiliseconds = this.slider.getAttribute('data-auto-rotate-time');
+    this.autoRotateEnabled = this.slider.getAttribute('data-auto-rotate-enabled');
+    this.hideAllSlides();
+    if (this.slides.length < 2) {
+      this.hideControls();
+    }
+    this.showSlide(0);
+    this.on();
+    this.initSwipe();
+  }
+
+  on() {
+    this.off();
+    this.controlLeftClicked = on(this.controlLeft, EVENTS.CLICK, this.handleControlLeftClicked);
+    this.controlRightClicked = on(this.controlRight, EVENTS.CLICK, this.handleControlRightClicked);
+  }
+
+  off() {
+    if (this.controlLeftClicked) {
+      this.controlLeftClicked();
+    }
+    if (this.controlRightClicked) {
+      this.controlRightClicked();
+    }
+  }
+
+  initSwipe() {
+    const swiper = new Swipe(this.slider);
+    swiper.onLeft(() => {
+      this.autoRotateEnabled = false;
+      this.showSlide(+1);
+    });
+    swiper.onRight(() => {
+      this.autoRotateEnabled = false;
+      this.showSlide(-1);
+    });
+    swiper.run();
+  }
+
+  handleControlLeftClicked = () => {
+    this.autoRotateEnabled = false;
+    this.showSlide(-1);
+  };
+
+  handleControlRightClicked = () => {
+    this.autoRotateEnabled = false;
+    this.showSlide(+1);
+  };
+
+  hideAllSlides() {
+    for (let i = 0, { length } = this.slides; i < length; i++) {
+      this.slides[i].style.display = 'none';
+    }
+  }
+
+  hideControls() {
+    this.controlRight.style.display = 'none';
+    this.controlLeft.style.display = 'none';
+  }
+
+  showSlide(positionDifference) {
```

Comment:
I'm sure it's not necessary to hide all slides by inline styles and for animation CSS transitions should be sufficient.

Reply:
The transitions are done this way because there are 2 transitions. One when the item is entering from right and one from left. Depending on which arrow/swipe is used we need to show different transition. 
let me know if you have better solution. I'll be happy to improve the existing solution.

Reply:
Alright I checked out the branch and started the app to see the animation.
It's fine for me like that, I just would not use CSS Animations if CSS Transitions do the same job.